### PR TITLE
make sure scratch build from PR finished

### PR DESCRIPTION
**Pull request by bgoncalv**
related to https://pagure.io/fedora-ci/general/issue/76

### CODE REVIEW

**Comment by bgoncalv** [comment]
@thrix could you check this?

**Comment by hroncok** [comment]
The logic seems good. Thanks!

File: config/Dockerfiles/fedoraci-runner/rpmbuild/koji_build_pr.sh [hunk 1]
```diff
@@ -61,15 +61,30 @@ koji ${KOJI_PARAMS} build --wait --arch-override=x86_64 --scratch ${branch} ${fe
...
-echo "status=SUCCESS" >> ${LOGDIR}/job.props
+
 popd
 
 SCRATCHID=$(cat ${LOGDIR}/kojioutput.txt | awk '/Created task:/ { print $3 }')
 echo "koji_task_id=${SCRATCHID}" >> ${LOGDIR}/job.props
 
+# Make sure koji build finished
+# https://pagure.io/fedora-ci/general/issue/76
+while koji taskinfo ${SCRATCHID} | grep "State: open"; do
```
Comment (hroncok) [inline config/Dockerfiles/fedoraci-runner/rpmbuild/koji_build_pr.sh]: this (taskinfo) could have a temporary error that leads to grep failure. also, the status could be "free". should this grep the other way around (i.e. continue to loop if it cannot grep closed|failed|cancelled?

File: config/Dockerfiles/fedoraci-runner/rpmbuild/koji_build_pr.sh [hunk 2]
```diff
@@ -61,15 +61,30 @@ koji ${KOJI_PARAMS} build --wait --arch-override=x86_64 --scratch ${branch} ${fe
...
-echo "status=SUCCESS" >> ${LOGDIR}/job.props
+
 popd
 
 SCRATCHID=$(cat ${LOGDIR}/kojioutput.txt | awk '/Created task:/ { print $3 }')
 echo "koji_task_id=${SCRATCHID}" >> ${LOGDIR}/job.props
 
+# Make sure koji build finished
+# https://pagure.io/fedora-ci/general/issue/76
+while koji taskinfo ${SCRATCHID} | grep "State:" | grep -Ev "closed|failed|cancelled"; do
```
Comment (hroncok) [inline config/Dockerfiles/fedoraci-runner/rpmbuild/koji_build_pr.sh]: when this greps failed...

File: config/Dockerfiles/fedoraci-runner/rpmbuild/koji_build_pr.sh [hunk 3]
```diff
@@ -61,15 +61,30 @@ koji ${KOJI_PARAMS} build --wait --arch-override=x86_64 --scratch ${branch} ${fe
...
+    # Set status if job fails to build the rpm
+    RPMBUILD_RC=$?
+    if [ "$RPMBUILD_RC" != 0 ]; then
+         echo "status=FAIL" >> ${LOGDIR}/job.props
+         echo -e "ERROR: KOJI BUILD\nSTATUS: $RPMBUILD_RC"
+         exit 1
+    fi
+done
+
+echo "status=SUCCESS" >> ${LOGDIR}/job.props
```
Comment (hroncok) [inline config/Dockerfiles/fedoraci-runner/rpmbuild/koji_build_pr.sh]: ...it jumps here and sets the result to success.

**Comment by thrix** [comment]
@bgoncalv seems legit to me also. so +1

**Comment by bgoncalv** [comment]
[merge]

**Comment by bgoncalv** [comment]
@hroncok thanks for the review :)

**Pull request merged**